src/main/java/org/fogbowcloud/manager/core/plugins/compute/openstack/OpenStackNovaV2ComputePlugin.java

```diff
@@ -0,0 +1,438 @@
+                EntityUtils.consume(response.getEntity());
+            } catch (Throwable t) {
+                // Do nothing
+            }
+        }
+
+        // delete message does not have message
+        checkStatusResponse(response, "");
+    }
+
+    protected String doPostRequest(String endpoint, Token localToken, JSONObject jsonRequest) throws RequestException {
+        LOGGER.debug("Doing POST request to OpenStack for creating an instance...");
+
+        HttpResponse response = null;
+        String responseStr;
+
+        try {
+            HttpPost request = new HttpPost(endpoint);
+            request.addHeader(RequestHeaders.CONTENT_TYPE.getValue(), RequestHeaders.JSON_CONTENT_TYPE.getValue());
+            request.addHeader(RequestHeaders.ACCEPT.getValue(), RequestHeaders.JSON_CONTENT_TYPE.getValue());
+            request.addHeader(RequestHeaders.X_AUTH_TOKEN.getValue(), localToken.getAccessId());
+
+            request.setEntity(new StringEntity(jsonRequest.toString(), StandardCharsets.UTF_8));
+            response = client.execute(request);
+            responseStr = EntityUtils.toString(response.getEntity(), StandardCharsets.UTF_8);
+        } catch (Exception e) {
+            LOGGER.error(e);
+            throw new RequestException(ErrorType.BAD_REQUEST, ResponseConstants.IRREGULAR_SYNTAX);
+        } finally {
+            try {
+                EntityUtils.consume(response.getEntity());
+            } catch (Throwable t) {
+                // Do nothing
+            }
+        }
+
+        checkStatusResponse(response, responseStr);
+
+        return responseStr;
+    }
+
+    protected JSONObject generateJsonRequest(String imageRef, String flavorRef, String userdata,
+                                             String keyName, String networkId) throws JSONException {
+        LOGGER.debug("Generating JSON to send as the body of instance POST request...");
+
+        JSONObject server = new JSONObject();
+        server.put(NAME_JSON_FIELD, FOGBOW_INSTANCE_NAME + UUID.randomUUID().toString());
+        server.put(IMAGE_JSON_FIELD, imageRef);
+        server.put(FLAVOR_JSON_FIELD, flavorRef);
+
+        if (userdata != null) {
+            server.put(USER_DATA_JSON_FIELD, userdata);
+        }
+
+        if (networkId != null && !networkId.isEmpty()) {
+            List<JSONObject> nets = new ArrayList<>();
+            JSONObject net = new JSONObject();
+            net.put(UUID_JSON_FIELD, networkId);
+            nets.add(net);
+            server.put(NETWORK_JSON_FIELD, nets);
+        }
+
+        if (keyName != null && !keyName.isEmpty()) {
+            server.put(KEY_JSON_FIELD, keyName);
+        }
+
+        JSONObject root = new JSONObject();
+        root.put(SERVER_JSON_FIELD, server);
+
+        return root;
+    }
+
+    private void checkStatusResponse(HttpResponse response, String message) throws RequestException {
+        if (response.getStatusLine().getStatusCode() == HttpStatus.SC_UNAUTHORIZED) {
```

Comment:
These ifs and else are looking bad, you could simplify these method by mapping the Status code with the ErrorType and ResponseConstants. Therefore this mapping looks to me that will be used by others classes that handles HTTP requests.

Reply:
`StatusResponseMap` created. It has a HashMap which will map `HttpStatus` with `StatusResponse`. 

Example: 
```java 
statusResponseMap.put(HttpStatus.SC_UNAUTHORIZED, new StatusResponse(ErrorType.UNAUTHORIZED, ResponseConstants.UNAUTHORIZED));
```